server/application/rest/work_shift.py

```diff
@@ -36,28 +37,69 @@
 @cross_origin()
 def work_shifts():
     """
-        On GET: The function returns a list of all work shifts in the system.
-        On POST: The function adds shifts to the system.
+    On GET: The function returns a list of all work shifts in the system.
+    On POST: The function adds shifts to the system.
     """
     if request.method == "GET":
         repo = MemRepo(shifts)
-        result = workshift_list_use_case(repo)
-        return Response(
-            json.dumps(result, cls=WorkShiftJsonEncoder),
-            mimetype="application/json",
-            status=200,
-        )
+        try:
+            result = workshift_list_use_case(repo)
+            response_data = json.dumps(result, cls=WorkShiftJsonEncoder)
+            return Response(
+                response_data,
+                mimetype="application/json",
+                status=200
+            )
+        except (AuthenticationError, NotFoundError) as e:
+            return handle_error_response(e)
     elif request.method == "POST":
         user = get_user_from_token(request.headers)
         data = request.get_json()
         for shift in data:
             shift["worker"] = user
         repo = MemRepo(shifts)
-        workshift_add_multiple_use_case(repo, data)
+        try:
+            workshift_add_multiple_use_case(repo, data)
+            response_data = json.dumps(data, cls=WorkShiftJsonEncoder)
+            return Response(
+                response_data,
+                mimetype="application/json",
+                status=200
+            )
+        except (AuthenticationError, NotFoundError) as e:
+            return handle_error_response(e)
+
+
+def get_user_from_token(headers):
+    if "Authorization" in headers:
+        return headers["Authorization"]
+    raise AuthenticationError("Authentication failed")
+
+
+@blueprint.route("/shifts/<shift_id>", methods=["DELETE"])
+@cross_origin()
+def delete_work_shift(shift_id):
+    try:
+        shift_id = str(shift_id)
+        user_email = get_user_from_token(request.headers)
+        repo = MemRepo(shifts)
+        delete_shift_use_case(repo, shift_id, user_email)
         return Response(
-            json.dumps(data, cls=WorkShiftJsonEncoder),
+            json.dumps({"message": "Shift deleted successfully"}),
             mimetype="application/json",
-            status=200,
+            status=200
         )
-def get_user_from_token(headers):
-    return headers["Authorization"]
+    except (AuthenticationError, NotFoundError) as e:
+        return handle_error_response(e)
+
+
+def handle_error_response(error):
```

Comment:
Again, please go back to chapter 5 of the book for how error responses are handled.